torch/utils/_pytree.py

```diff
@@ -672,9 +675,12 @@ class TreeSpec:
     num_children: int = dataclasses.field(init=False)
 
     def __post_init__(self) -> None:
-        self.num_nodes = 1 + sum(spec.num_nodes for spec in self.children_specs)
-        self.num_leaves = sum(spec.num_leaves for spec in self.children_specs)
-        self.num_children = len(self.children_specs)
+        num_nodes = 1 + sum(spec.num_nodes for spec in self.children_specs)
+        num_leaves = sum(spec.num_leaves for spec in self.children_specs)
+        num_children = len(self.children_specs)
+        object.__setattr__(self, "num_nodes", num_nodes)
```

Comment:
Hmm, not sure if dynamo can trace through this hack to `__setattr__` properly.

Reply:
I ran pytree test with dynamo locally and all tests passed.

According to the CI result, seems dynamo is not fully support `object.__setattr__`. Here is the previous PR and issue tracker.

- #114964
- https://github.com/pytorch/pytorch/pull/114946#discussion_r1412413180